cobbler/items/abstract/inheritable_item.py

```diff
@@ -0,0 +1,339 @@
+            return self.get_conceptual_parent()
+        return parent
+
+    @property
+    def children(self) -> List["ITEM_UNION"]:
+        """
+        The list of logical children of any depth.
+        :getter: An empty list in case of items which don't have logical children.
+        :setter: Replace the list of children completely with the new provided one.
+        """
+        results: List[Any] = []
+        list_items = self.api.get_items(self.COLLECTION_TYPE)
+        for obj in list_items:
+            if obj.get_parent == self._name:
+                results.append(obj)
+        return results
+
+    def tree_walk(self) -> List["ITEM_UNION"]:
+        """
+        Get all children related by parent/child relationship.
+        :return: The list of children objects.
+        """
+        results: List[Any] = []
+        for child in self.children:
+            results.append(child)
+            results.extend(child.tree_walk())
+
+        return results
+
+    @property
+    def descendants(self) -> List["ITEM_UNION"]:
+        """
+        Get objects that depend on this object, i.e. those that would be affected by a cascading delete, etc.
+
+        .. note:: This is a read only property.
+
+        :getter: This is a list of all descendants. May be empty if none exist.
+        """
+        childs = self.tree_walk()
+        results = set(childs)
+        childs.append(self)  # type: ignore
+        for child in childs:
+            for item_type in self.TYPE_DEPENDENCIES[child.COLLECTION_TYPE]:
+                dep_type_items = self.api.find_items(
+                    item_type[0], {item_type[1]: child.name}, return_list=True
+                )
+                if dep_type_items is None or not isinstance(dep_type_items, list):
+                    raise ValueError("Expected list to be returned by find_items")
+                results.update(dep_type_items)
+                for dep_item in dep_type_items:
+                    results.update(dep_item.descendants)
+        return list(results)
+
+    @LazyProperty
+    def is_subobject(self) -> bool:
+        """
+        Weather the object is a subobject of another object or not.
+
+        :getter: True in case the object is a subobject, False otherwise.
+        :setter: Sets the value. If this is not a bool, this will raise a ``TypeError``.
+        """
+        return self._is_subobject
+
+    @is_subobject.setter
+    def is_subobject(self, value: bool) -> None:
+        """
+        Setter for the property ``is_subobject``.
+
+        :param value: The boolean value whether this is a subobject or not.
+        :raises TypeError: In case the value was not of type bool.
+        """
+        if not isinstance(value, bool):  # type: ignore
+            raise TypeError(
+                "Field is_subobject of object item needs to be of type bool!"
+            )
+        self._is_subobject = value
+
+    def grab_tree(self) -> List[Union["ITEM", "Settings"]]:
```

Comment:
Can we fix these? Now, it should be `List[Union["InheritableItem", "Settings"]]`, right? Not sure if other parts of the code base have the same issue.